libraries/GCS_MAVLink/MAVLink_routing.cpp

```diff
@@ -130,10 +132,12 @@ bool MAVLink_routing::check_and_forward(GCS_MAVLINK &in_link, const mavlink_mess
         return true;
     }
 
+#if HAL_ADSB_ENABLED
     if (msg.msgid == MAVLINK_MSG_ID_ADSB_VEHICLE) {
```

Comment:
I think this would work at runtime for tracker.
```suggestion
    if ((msg.msgid == MAVLINK_MSG_ID_ADSB_VEHICLE) && (AP::ADSB() != nullptr)) {
```

I guess we could do a step further and also check that ADSB_TYPE is actually set to something.

Reply:
I have implemented the singleton check, but not that type is set. We could add `&& AP::ADSB()->enabled()`.

Reply:
yes, enabled check is needed I think